### ISSUE TITLE
Feature Request - Geometry tool icons for shapes

### ISSUE THREAD

**Issue by turtletooth** [opentoonz/opentoonz#665]
I know this is a small thing, but I think it would make the geometry tool more accessible.  Instead of the dropdown that currently exists:
![shape](https://cloud.githubusercontent.com/assets/4576381/17039447/cbd4e5c2-4f57-11e6-9132-40be57b6baba.jpg)
Could there be icons for each type of shape?  This would reduce the number of clicks and give a more visual representation of what the user is about to use.

I also think the geometry tool icon itself doesn't very clearly explain what the tool is for.  It looks like it is for measurement.

**Comment by Rayek** [comment]
Very much in support of this. The dropdown lists are quite a bad choice in the first place as far as interaction methods are concerned.

Btw, I wish the edit tool widget would be restored, because that dropdown, which controls the edit tool mode, is quite terrible. In Toonz a nice all-in-one widget is available. I really wish someone would restore that control widget - interestingly enough the new icons (svg format instead of png) are all available in the source code assets, including the one for the combo edit tool.

**Comment by shun-iwasawa** [comment]
@Rayek Reverting edit tool gadget is being discussed in #157 . Please take a look.

**Comment by nittolo** [comment]
How about splitting the geometric tool up into separate tools instead? Right now, it seems like the Geometric Tool is just a bunch of random tools mashed into one, and if it were split up, you could simply switch between the new tools with keyboard shortcuts. There could be a shape tool (Rectangle, Polygon, Circle, and Ellipse), line tool (Line and Arc), and pen tool (just Polyline). Alternatively, there could be a polygon tool (Rectangle and Polygon), line tool (Line, Arc, and Polyline), and circle tool (Circle and Ellipse). There are many different ways to split it up. I'm not sure which would be most efficient.

Another option is to just add keyboard shortcuts for each mode, as is done for drop down menus like Mode and Type.

Regardless of actually changing the functionality of the tool, I definitely agree that the icon should be changed.

**Comment by blurymind** [comment]
I dont like the splitting into tools idea because the toolbox is already
running out of space.
I dont like the keybinding idea because that will make the tool modes
hidden to the user

**Comment by nittolo** [comment]
@blurymind keybindings wouldn't make them hidden... the Mode and Type menus (like on the Fill Tool) are still shown but you can also have keyboard shortcuts to each option.

**Comment by Rayek** [comment]
Please, get rid of those drop-down menus. Expose the sub-tools as icons/buttons in the toolbar instead. And I have nothing against custom keybindings (which I would prefer, and advanced users probably as well) - but do add those visual buttons in the toolbar.

This is also important for Cintiq/tablet/Wacom users.

**Comment by blurymind** [comment]
I propose to borrow from blender in this case:
https://github.com/opentoonz/opentoonz/issues/157#issuecomment-234671470

![blenderexamplewidget](https://cloud.githubusercontent.com/assets/6495061/17073428/af130966-5068-11e6-8bd4-060948360d49.gif)

And please bring back the combined widget as an option. Im not the only one requesting it

**Comment by turtletooth** [comment]
I'm on it.  I'll start here to see how the edit tool discussion plays out.  Perhaps @shun-iwasawa will want to do that himself.

**Comment by gab3d** [comment]
This discussion was very interesting...

I know the idea of changing tool mode/type/axis by shortcuts was finally implemented, and it is indeed an important thing, usability wise.
But the idea of replacing some of the dropdown menus in favor of icons was a nice one nice too.

Also, the idea of giving the Geometric tool Polyline mode a "tool status" (Pen tool) would certainly improve new users' experience with the program.
Currently it seems (to the eyes of new uninstructed users) that OT doesn't have such a basic vector tool as a Pen tool, just because it's really 'hidden' inside a tool mode dropdown menu, inside a tool whose name (Geometric tool) doesn't give a clue of containing such a possibility.

I know toolbar can get a little more crowded by this, but it's only one more icon for a very basic tool, users of any other program expect to have visible to the "naked eye".

Just my two cents on this...
Thanks for listening!

**Comment by RodneyBaker** [comment]
@gab3d  I"ll add my 2 cents... for what little it is worth and we are well on our way to a nickel.

I agree the Polyline tool can useful but to warrant a place of it's own on the main tool menu...
Is the polyline tool really used that much?  
I anticipate you'll say 'yes'.

I have been trying to consider using it more but must confess I've rarely used it in any software.
My work likely suffers for this.

Despite my lack of affection for the Polylinetool, your question does have merit.
Previously Toonz/OpenToonz was not seen to any great degree as Illustration software.
It has been gaining some serious ground in this capacity of late.

This is a closed topic so I don't see why the discussion cannot continue here (short of it being a distraction to the developers but... sometimes even they need a break!) ;)

In short, you've got me wondering what an optimized Drawing Room (to optimize digital workflow) might look like.

**Comment by artisteacher** [comment]
The polyline mode of the Geometric tool almost has the complete functionality of a pen tool - it just lacks the flexibility of changing between all anchor types while drawing. If it matters, Flash has a pen tool.  Personally, I'd love to see polyline separated out as a pen tool. It makes always fillable shapes & can minimize vector data - I think those are strong arguments!  

If that happens, a pen tool really needs to have the ability to convert the anchor to a cusp point (anchor point with 2 independent handles) while drawing. Usually this is done by holding down on alt _after_ clicking and dragging (to adjust the curve of the first handle) to then adjust the second handle independently,  _before_ releasing the click. This sets up the next curve and saves a lot of time. 

Right now you can easily change between curved and straight lines while drawing with polyline, but the curved handles are linked. While you can convert the anchor type with the control point editor, it's not nearly as efficient and usually means an extra 2 steps (1 - converting the anchor type &  2 - fixing the curves to be what was originally intended) every time.
